internal/npm_install/generate_pnp_build_files.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from "fs";
+import * as path from "path";
+
+const workspacePath = process.argv.slice(2)[0];
+
+function mkdirp(p: string) {
+  if (!fs.existsSync(p)) {
+    mkdirp(path.dirname(p));
+    fs.mkdirSync(p);
+  }
+}
+
+function writeFileSync(p: string, content: string) {
+  mkdirp(path.dirname(p));
+  fs.writeFileSync(p, content);
+}
+
+function createBuildFile(pkg: string) {
+  const contents = `package(default_visibility = ["//visibility:public"])
+load("@build_bazel_rules_nodejs//internal/npm_install:node_module_library.bzl", "node_module_library")
+
+filegroup(
+    name = "${pkg}",
+    srcs = [] # we dont care about the actual content, we want a list of dependencies to modify the pnp file around
+)`;
+
+  writeFileSync(`${pkg}/BUILD.bazel`, contents);
+}
+
+async function main() {
+  const packageJson = require(path.join(workspacePath, "package.json"));
+  const dependencies = Object.keys(packageJson.dependencies || {});
+  const devDependencies = Object.keys(packageJson.dependencies || {});
+  dependencies.concat(devDependencies).forEach(createBuildFile);
+  const contents = `package(default_visibility = ["//visibility:public"])
+exports_files([
+    ".pnp.js",
+    "package.json",
+])`;
+  ``;
+  writeFileSync(`BUILD.bazel`, contents);
+}
+
+main();
```

Comment:
inject `process.argv` here:

```suggestion
main(process.argv[3]);
```

so that the file can be required without side-effects or potentical errors.

Reply:
or ideally call main from a CLI entry file.